### PR TITLE
feat(macros/Compat): Import `browser‑compat‑toolkit` changes

### PR DESCRIPTION
**Pull request by ExE-Boss**
## Split&nbsp;into:
- Fix&nbsp;typos and&nbsp;whitespace: https://github.com/mdn/kumascript/pull/1084 (merged)
- Improve `mdn_url` handling: https://github.com/mdn/kumascript/pull/1085
- Fix stray closing `</tr>` tag: https://github.com/mdn/kumascript/pull/1086
- Rewrite `writeFlagsNote(…)` function: https://github.com/mdn/kumascript/pull/1087

---

This PR imports the following [browser-compat-toolkit](https://github.com/mdn/browser-compat-toolkit) changes that I made:

- [x] https://github.com/mdn/browser-compat-toolkit/pull/7 (moved&nbsp;to https://github.com/mdn/kumascript/pull/1086)
- [x] https://github.com/mdn/browser-compat-toolkit/pull/5 (moved&nbsp;to https://github.com/mdn/kumascript/pull/1085)
- [x] https://github.com/mdn/browser-compat-toolkit/pull/15 (moved&nbsp;to https://github.com/mdn/kumascript/pull/1087)

## Features blocked by&nbsp;this:
- [x] Unhard&#x2011;code preferences using `pref_url` (https://github.com/mdn/browser-compat-data/pull/3407; moved&nbsp;to https://github.com/mdn/kumascript/pull/1087)
- [ ] Add support for `<=`&nbsp;syntax (https://github.com/mdn/browser-compat-data/issues/3021)

## See also:
- https://github.com/mdn/browser-compat-data/issues/2354

---

review?(@a2sheppy, @chrisdavidmills, @davidflanagan, @Elchi3, @escattone, @jwhitlock, @MatonAnthony, @wbamberg)

### CODE REVIEW

**Comment by ExE-Boss** [comment]
review?(@jwhitlock, @wbamberg): Can&nbsp;we&nbsp;please get&nbsp;this&nbsp;merged?

File: macros/Compat.ejs [hunk 1]
```diff
@@ -264,92 +268,84 @@ function getSupportClass(supportInfo) {
...
+ *
+ * @param {object} supportData is a support_statement
+ * @param {string} browserId is a compat_block browser ID
+ *
+ * @return {string} The note for the flags information.
+ */
+function writeFlagsNote (supportData, browserId) {
+  // TODO: Move these to the mdn-browser-compat-data package.
+  const firefoxPrefs = 'about:config';
+  const chromePrefs = 'chrome://flags';
```
Comment (ExE-Boss) [inline macros/Compat.ejs]: See&nbsp;https://github.com/mdn/browser-compat-data/pull/3407 for&nbsp;unhard&#x2011;coding&nbsp;this.

File: macros/Compat.ejs [hunk 2]
```diff
@@ -37,6 +37,7 @@ var output = '';
 var category = query.split(".")[0];
 var legendItems = new Set(); // entries will be unique
 
+// TODO: Migrate localization to Fluent ( https://www.npmjs.com/package/fluent )
```
Comment (Elchi3) [inline macros/Compat.ejs]: Please remove this TODO. I don't think we decided to use Fluent. A decision likely requires an ADR. https://github.com/mdn/mdn/tree/master/ADRs

**Comment by ExE-Boss** [comment]
Number&nbsp;6 is&nbsp;a&nbsp;result of&nbsp;whitespace changes, which I’ve moved to&nbsp;https://github.com/mdn/kumascript/pull/1084.

**Comment by ExE-Boss** [comment]
I’ve now split the changes into the following PRs:
- https://github.com/mdn/kumascript/pull/1084
- https://github.com/mdn/kumascript/pull/1085
- https://github.com/mdn/kumascript/pull/1086
- https://github.com/mdn/kumascript/pull/1087

**Comment by Elchi3** [comment]
Thank you!

**Pull request closed without merging**